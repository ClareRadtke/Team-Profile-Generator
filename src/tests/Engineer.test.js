const { Engineer } = require("../lib/classes");

describe(Engineer, () => {
  it("should create a new Employee instance", () => {
    const engineer = new Engineer(
      "John Doe",
      "123",
      "foo@example.com",
      "JohnDoe"
    );
    expect(engineer).toMatchSnapshot();
  });
  describe("getName()", () => {
    it("should return the name", () => {
      const engineer = new Engineer(
        "John Doe",
        "123",
        "foo@example.com",
        "JohnDoe"
      );
      expect(engineer.getName()).toBe("John Doe");
    });
  });
  describe("getId()", () => {
    it("should return the id", () => {
      const engineer = new Engineer(
        "John Doe",
        "123",
        "foo@example.com",
        "JohnDoe"
      );
      expect(engineer.getId()).toBe("123");
    });
  });
  describe("getEmail()", () => {
    it("should return the email", () => {
      const engineer = new Engineer(
        "John Doe",
        "123",
        "foo@example.com",
        "JohnDoe"
      );
      expect(engineer.getEmail()).toBe("foo@example.com");
    });
  });
  describe("getRole()", () => {
    it("should return the role", () => {
      const engineer = new Engineer(
        "John Doe",
        "123",
        "foo@example.com",
        "JohnDoe"
      );
      expect(engineer.getRole()).toBe("Engineer");
    });
  });
  describe("getGithub()", () => {
    it("should return the Github username", () => {
      const engineer = new Engineer(
        "John Doe",
        "123",
        "foo@example.com",
        "JohnDoe"
      );
      expect(engineer.getGithub()).toBe("JohnDoe");
    });
  });
});
