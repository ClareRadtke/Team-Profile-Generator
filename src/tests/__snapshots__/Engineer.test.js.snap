// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engineer should create a new Employee instance 1`] = `
Engineer {
  "email": "foo@example.com",
  "github": "JohnDoe",
  "id": "123",
  "name": "John Doe",
}
`;
